prydaen-favors.lic

```diff
@@ -0,0 +1,114 @@
+
+    Must have at least 65 ranks of perception to get to the Prydaen boulder.
+
+    **WARNING**
+    - Be sure your health_threshold is not going to cause you to log off.
+      You will have significant blood loss doing Eu's offering.
+    
+    changelog:
+        2.0 (2022-01-30):
+            **Functional with checkfavors or by itself (waiting on merge)
+            Purchases rat-pelt leather from Outfitting Society to use as an offering
+=end
+
+custom_require.call(%w[common common-travel common-money])
+
+class Prydaen_Favors
+  include DRC
+  include DRCT
+
+  def initialize
+    unless DRStats.race == 'Prydaen'
+      echo "No meow."
+      exit
+    end
+    unless DRSkill.getrank('Perception') > 64
+      echo "Not enough perception to get to the Prydaen boulder."
+      exit
+    end
+    settings = get_settings
+    check_offering
+  end
+
+  def check_offering
+    if /could not/ =~ bput("tap my rat-pelt leather", 'You tap', 'I could not find')
+      get_coin
+    end
+    pray_boulder
+  end
+
+  def get_coin
+    walk_to('1900')
+    fput('withdraw 2 silver')
+    order_item
+  end
+
+  def order_item
+    walk_to('16667')
+    return if DRC.bput("order 14", 'Just order it again', 'you don\'t have enough coins') == 'you don\'t have enough coins'
+    DRC.bput("order 14", 'takes some coins from you')
+  end
+
+  def pray_boulder
+    walk_to('7306')
+    case bput('pray boulder', /prayer to the Triquetra on your lips/, /nearby bush sprouts a new twig complete with leaves/, /nearby bush has begun to rapidly die/, /the blossom is the only proof/)
+    when /nearby bush sprouts a new twig complete with leaves/
+      place_item
+    when /nearby bush has begun to rapidly die/
+      place_item
+    when /the blossom is the only proof/
+      get_blossom
+    when /prayer to the Triquetra on your lips/
+      echo 'You already did this.  Complete your previous offering.'
+      exit
+    end
+  end
+
+  def get_blossom
+    case bput('get blossom', /You pick up a heartsblood trillium blossom./)
+    when /You pick up a heartsblood trillium blossom./
+      fput('put blossom on boulder')
+      waitfor 'you see that the blossom has become'
+      get_symbol
+    end
+  end
+
+  def place_item
+    case bput('get my rat-pelt leather', /You get/, /You carefully remove/, /What were you referring to?/, /You are already holding that./)
+    when /You get/, /You carefully remove/, /You are already holding that./
+      fput('put rat-pelt leather on boulder')
+      waitfor 'You reverently place'
+      fput('get symbol')
+      invoke_symbol
+    when /What were you referring to?/
+      get_coin
+    end
+  end
+
+  def get_symbol
+    fix_standing
+    fput('get symbol')
+    invoke_symbol
+  end
+
+  def invoke_symbol
+    case bput('invoke my symbol', /With a flash your .+ vanishes/, /You attempt to invoke/, /feel a part of yourself flow into/)
+    when /With a flash your .+ vanishes/
+      stow_hands
```

Comment:
If all cases take the same action, can we drop the `case` logic and just always stow hands and exit?